iroh/src/endpoint.rs

```diff
@@ -1295,15 +1289,12 @@ impl Connecting {
     /// Extracts the ALPN protocol from the peer's handshake data.
     // Note, we could totally provide this method to be on a Connection as well.  But we'd
     // need to wrap Connection too.
-    pub async fn alpn(&mut self) -> Result<Vec<u8>> {
-        let data = self.handshake_data().await?;
-        match data.downcast::<quinn::crypto::rustls::HandshakeData>() {
-            Ok(data) => match data.protocol {
-                Some(protocol) => Ok(protocol),
-                None => bail!("no ALPN protocol available"),
-            },
-            Err(_) => bail!("unknown handshake type"),
-        }
+    pub async fn alpn(&mut self) -> Result<Vec<u8>, ConnectionError> {
+        let data = self.inner.handshake_data().await?;
+        let data = data
+            .downcast::<quinn::crypto::rustls::HandshakeData>()
+            .expect("crypto setup for iroh");
+        Ok(data.protocol.expect("ALPN required by iroh crypto setup"))
     }
```

Comment:
```suggestion
    pub async fn alpn(&mut self) -> Result<Vec<u8>, ConnectionError> {
        let data = self.inner.handshake_data().await?;
        let data = data
            .downcast::<quinn::crypto::rustls::HandshakeData>()
            .expect("invariant checked by iroh's custom rustls verifier");
        Ok(data.protocol.expect("invariant upheld by iroh's custom rustls verifier"))
    }
```
Ideally insert the function names of whatever is *actually* verifying these invariants here.